packages/scripts/test/optimistic-oracle-umip/3_Verify.ts

```diff
@@ -0,0 +1,61 @@
+// This script verify that the upgrade was executed correctly. It can be
+// run on a local hardhat node fork of the mainnet or can be run directly on the mainnet to execute the upgrade
+// transactions. To run this on the localhost first fork mainnet into a local hardhat node by running:
+// HARDHAT_CHAIN_ID=1 yarn hardhat node --fork https://mainnet.infura.io/v3/<YOUR-INFURA-KEY> --port 9545 --no-deploy
+// Then execute the script from core:
+// OPTIMISTC_ORACLE_V2=<OPTIMISTC_ORACLE_V2_ADDRESS> yarn workspace @uma/scripts hardhat run ./test/optimistic-oracle-umip/3_Verify.ts --network localhost
+
+const hre = require("hardhat");
+
+const assert = require("assert").strict;
+
+const { RegistryRolesEnum } = require("@uma/common");
+
+import { Governor, Finder, Registry } from "@uma/core/contract-types/ethers";
+
+const { getAddress } = require("@uma/contracts-node");
+
+const OPTIMISTIC_ORACLE_V2 = "OptimisticOracleV2"; // TODO use interfaceName.OptimisticOracle
+const deployed_optimistic_oracle_address = process.env["OPTIMISTC_ORACLE_V2"];
+
+const getContractInstance = async <T>(contractName: string): Promise<T> => {
+  const networkId = await hre.getChainId();
+  const factory = await hre.ethers.getContractFactory(contractName);
+  const contractAddress = await getAddress(contractName, Number(networkId));
+  return (await factory.attach(contractAddress)) as T;
+};
+
+async function main() {
+  if (!deployed_optimistic_oracle_address) throw new Error("OPTIMISTC_ORACLE_V2 environment variable not set");
+
+  const finder = await getContractInstance<Finder>("Finder");
+  const governor = await getContractInstance<Governor>("Governor");
+  const registry = await getContractInstance<Registry>("Registry");
+
+  console.log("Verifying that Governor doesn't hold the creator role...");
+  !(await registry.holdsRole(RegistryRolesEnum.CONTRACT_CREATOR, governor.address));
+  console.log("Verified!");
+
+  console.log("Verifying that the OptimisticOracle is registered with the Registry...");
+  assert(await registry.isContractRegistered(deployed_optimistic_oracle_address));
+  console.log("Verified!");
+
+  console.log("Verifying that the OptimisticOracleV2 is registered with the Finder...");
+  assert.equal(
+    (await finder.getImplementationAddress(hre.ethers.utils.formatBytes32String(OPTIMISTIC_ORACLE_V2))).toLowerCase(),
+    deployed_optimistic_oracle_address.toLowerCase()
+  );
+  console.log("Verified!");
```

Comment:
Might be worth checking that the old Optimistic oracle is still in the finder under the old name and registered with the registry?